src/preloader.js

```diff
@@ -88,7 +89,15 @@ module.exports = function installPreloader(logger) {
     }
 
     if (takeProfile) {
-      benchmarkGeneration(logger, 'sampling profile', generateHeapSamplingProfile, profilerOptions, onToolEnd)
+      const controller = new AbortController()
+      profilerOptions.signal = controller.signal
+      const abort = controller.abort.bind(controller)
```

Comment:
Be sure to check to see if the `AbortSignal` has already been tripped before proceeding.

Reply:
done